assets/src/api.ts

```diff
@@ -66,6 +67,37 @@ export const apiCall = <T>({
       }
     })
 
+export const checkedApiCall = <T, U>({
+  url,
+  dataStruct,
+  parser,
+  defaultResult,
+  fetchArgs,
+}: {
+  url: string
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  defaultResult?: U
+  fetchArgs?: RequestInit
+}): Promise<U> =>
+  fetch(url, fetchArgs)
+    .then(checkResponseStatus)
+    .then(
+      (response) =>
+        parseJson(response) as Record<string, unknown> & { data: unknown }
```

Comment:
question: I may be missing  something about this - wouldn't `Record<string, unknown> &  { data: unknown }` evaluate to `{data: unknown}`?

Reply:
You're correct, I'm just so used to strongly-typed languages where the "record with field A of type B" type means that it _only_ has that field, not that it has at least that field.